myuw/dao/calendar_mapping.py

```diff
@@ -12,40 +15,28 @@
 
 
 def get_calendars_for_current_user(request):
-    return _get_calendars(request, _get_enrollments(request))
+
+    enrollments = {
+        'majors': [maj.major_name for maj in get_majors(request)['rollup']],
+        'minors': [mino.short_name for mino in get_minors(request)['rollup']]
+    }
```

Comment:
The original code only loop through the enrollments once to get both major and minor